lib/windowsexec/syscall_windows.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package windowsexec
+
+// this package is called windowsexec rather than just `windows` because
+// calling the package `windows` causes `mkwinsyscall` to generate code
+// without the appropriate package name.
+
+import (
+	"strings"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/sys/windows"
+)
+
+//go:generate go run golang.org/x/sys/windows/mkwinsyscall -output zsyscall_windows.go syscall_windows.go
+//sys	shellExecuteExW(info *shellExecuteInfoW) (wasSuccess bool) = shell32.ShellExecuteExW
+
+// shellExecuteInfoW is the input/output struct for ShellExecuteExW.
+// See the docs for information about the fields:
+// https://learn.microsoft.com/en-us/windows/win32/api/shellapi/ns-shellapi-shellexecuteinfow
+type shellExecuteInfoW struct {
+	cbSize         uint32
+	fMask          uint32
+	hwnd           syscall.Handle
```

Comment:
Nit: prefer `windows` over `syscall` whenever possible as `syscall` is deprecated: https://pkg.go.dev/syscall